pkg/autoscaler/autoscaler_test.go

```diff
@@ -113,3 +114,141 @@ func TestAutoscaler_Awake(t *testing.T) {
 		return latestRequest != 1
 	}, downscaleStabilizationPeriod, time.Second)
 }
+
+func TestAutoscaler_MinReplicas(t *testing.T) {
+	t.Parallel()
+	log := newLogger(t)
+
+	mux := sync.RWMutex{}
+	var latestRequest int32
+
+	minReplicas := int32(5)
+	maxReplicas := int32(10)
+
+	scalerMock := &ScalerFunc{
+		ScaleFunc: func(apiName string, request int32) error {
+			mux.Lock()
+			defer mux.Unlock()
```

Comment:
Robert: is this required for the test?

Reply:
yes because there are 2 goroutines accessing the variable